src/context/WalletProvider/WalletProvider.tsx

```diff
@@ -525,6 +526,65 @@ export const WalletProvider = ({ children }: { children: React.ReactNode }): JSX
           // case KeyManager.Ledger:
           // const ledgerWallet = await state.adapters.get(KeyManager.Ledger)?.[0].pairDevice()
           // return ledgerWallet
+          case KeyManager.Ledger: {
+            const featureFlags = preferences.selectors.selectFeatureFlags(store.getState())
+            const isLedgerReadOnlyEnabled = featureFlags.LedgerReadOnly
+
+            // This flag ensures we restore Ledger as connected in terms of state, i.e
+            // wether or not it is *physically* connected
+            if (isLedgerReadOnlyEnabled) {
+              try {
+                // Get the adapter again in each switch case to narrow down the adapter type
+                const ledgerAdapter = await getAdapter(localWalletType)
+
+                if (ledgerAdapter) {
+                  currentAdapters[localWalletType] = ledgerAdapter
+                  dispatch({ type: WalletActions.SET_ADAPTERS, payload: currentAdapters })
+                  // Fixes issue with wallet `type` being null when the wallet is loaded from state
+                  dispatch({
+                    type: WalletActions.SET_CONNECTOR_TYPE,
+                    payload: { modalType: localWalletType, isMipdProvider: false },
+                  })
+                }
+
+                // Don't call pairDevice() here as it requires user gesture for WebUSB permission
+                // Instead, restore the wallet state in read-only mode
+                const { name, icon } = SUPPORTED_WALLETS[KeyManager.Ledger]
+
+                dispatch({
+                  type: WalletActions.SET_WALLET,
+                  payload: {
+                    wallet: null, // No physical wallet connection yet
```

Comment:
TBD in a follow-up, most likely able to restore this too - and connect as "Disconnected"